spec/cli_spec.rb

```diff
@@ -22,18 +22,40 @@
   end
 
   context 'login' do
-    it 'exits with usage error if url missing' do
+    before(:each) do
+      stub_const("SessionAdapter::FILE_LOCATION", 'spec/tmp/.ccli/session' )
+    end
+
+    it 'exits with usage error if args missing' do
       set_command(:login)
 
+      expect(Kernel).to receive(:exit).with(usage_error_code)
+      allow_any_instance_of(NilClass).to receive(:split).and_return(["a", "b"])
+      expect{ subject.run }
+        .to output(/Credentials missing/)
+        .to_stderr
+    end
+
+    it 'exits with usage error if url missing' do
+      set_command(:login, 'WEj2eCJnwKbjw@')
+
       expect(Kernel).to receive(:exit).with(usage_error_code)
       expect{ subject.run }
         .to output(/URL missing/)
         .to_stderr
     end
 
-    it 'exits successfully when url given' do
-      set_command(:login, 'https://cryptopus.specs.com')
-      stub_const("SessionAdapter::FILE_LOCATION", 'spec/tmp/.ccli/session' )
+    it 'exits with usage error if token missing' do
+      set_command(:login, '@https://cryptopus.specs.com')
```

Comment:
always use example.com URL for test data.
example.com domain is reserved for this purpose ! specs.com exists and this could lead to unexpected behavior

-> cryptopus.example.com